pkg/storage/minio/checker.go

```diff
@@ -19,15 +18,25 @@ func (v *storageImpl) Exists(ctx context.Context, module, version string) (bool,
 	defer span.End()
 	versionedPath := v.versionLocation(module, version)
 	modPath := fmt.Sprintf("%s/go.mod", versionedPath)
-	_, err := v.minioClient.StatObject(v.bucketName, modPath, minio.StatObjectOptions{})
+	infoPath := fmt.Sprintf("%s/%s.info", versionedPath, version)
+	zipPath := fmt.Sprintf("%s/source.zip", versionedPath)
 
-	if minio.ToErrorResponse(err).Code == minioErrorCodeNoSuchKey {
-		return false, nil
-	}
+	var count int
+	objectCh, _ := v.minioCore.ListObjectsV2(v.bucketName, versionedPath, "", false, "", 0, "")
```

Comment:
any reason for not handling the error here?

Reply:
no, it should be handled.